src/components/users/LoginForm.js

```diff
@@ -17,31 +20,28 @@ const LoginForm = () => {
   const [inputs, setInputs] = useState({ email: '', password: '' });
   const { email, password } = inputs;
 
-  const [cleanAlert, setCleanAlert] = useState(false);
-
-  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [errors, setErrors] = useState({});
 
   const { loading, requestError, errorMsg } = useSelector(
     (state) => state.user
   );
 
-  const showLoginAlert = isSubmitted && requestError && !cleanAlert;
-
   const handleChange = ({ target: { name, value } }) => {
-    if (!loading) {
-      setInputs((inputs) => ({ ...inputs, [name]: value }));
-      setCleanAlert(true);
+    if (requestError) {
+      dispatch({ type: userConstants.USER_CLEAN_ALERT });
     }
+    setErrors(omit(errors, name));
+    setInputs((inputs) => ({ ...inputs, [name]: value }));
   };
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    setIsSubmitted(true);
-    if (email && password) {
-      setCleanAlert(false);
+    var currentErrors = validate(inputs, loginConstraints) || {};
```

Comment:
any reason to use **var** variables? I would prefer use ECMA5 variables(const, let) as long as possible.